stl/inc/chrono

```diff
@@ -617,27 +669,1288 @@ namespace chrono {
+            return _Year.ok() && _Month.ok();
+        }
+
+    private:
+        _CHRONO year _Year;
+        _CHRONO month _Month;
+    };
+
+    _NODISCARD constexpr bool operator==(const year_month& _Left, const year_month& _Right) noexcept {
+        return _Left.year() == _Right.year() && _Left.month() == _Right.month();
+    }
+    _NODISCARD constexpr strong_ordering operator<=>(const year_month& _Left, const year_month& _Right) noexcept {
+        const auto _Comp = _Left.year() <=> _Right.year();
+        if (_Comp != 0) {
+            return _Comp;
+        }
+
+        return _Left.month() <=> _Right.month();
+    }
+
+    _NODISCARD constexpr year_month operator+(const year_month& _Left, const months& _Right) noexcept {
+        const auto _Mo  = static_cast<long long>(static_cast<unsigned int>(_Left.month())) + (_Right.count() - 1);
+        const auto _Div = (_Mo >= 0 ? _Mo : _Mo - 11) / 12;
+        return year_month{_Left.year() + years{_Div}, month{static_cast<unsigned int>(_Mo - _Div * 12 + 1)}};
+    }
+    _NODISCARD constexpr year_month operator+(const months& _Left, const year_month& _Right) noexcept {
+        return _Right + _Left;
+    }
+
+    _NODISCARD constexpr year_month operator-(const year_month& _Left, const months& _Right) noexcept {
+        return _Left + -_Right;
+    }
+
+    _NODISCARD constexpr months operator-(const year_month& _Left, const year_month& _Right) noexcept {
+        return _Left.year() - _Right.year()
+               + months{static_cast<int>(static_cast<unsigned int>(_Left.month()))
+                        - static_cast<int>(static_cast<unsigned int>(_Right.month()))};
+    }
+
+    _NODISCARD constexpr year_month operator+(const year_month& _Left, const years& _Right) noexcept {
+        return {year{_Left.year() + _Right}, _Left.month()};
+    }
+
+    _NODISCARD constexpr year_month operator+(const years& _Left, const year_month& _Right) noexcept {
+        return _Right + _Left;
+    }
+
+    _NODISCARD constexpr year_month operator-(const year_month& _Left, const years& _Right) noexcept {
+        return _Left + -_Right;
+    }
+
+    constexpr year_month& year_month::operator+=(const months& _Months) noexcept {
+        *this = *this + _Months;
+        return *this;
+    }
+    constexpr year_month& year_month::operator-=(const months& _Months) noexcept {
+        *this = *this - _Months;
+        return *this;
+    }
+    constexpr year_month& year_month::operator+=(const years& _Years) noexcept {
+        *this = *this + _Years;
+        return *this;
+    }
+    constexpr year_month& year_month::operator-=(const years& _Years) noexcept {
+        *this = *this - _Years;
+        return *this;
+    }
+
+    _NODISCARD constexpr inline day _Last_day(const year& _Year, const month& _Month) {
+        constexpr day _Days[] = {
+            day{31}, day{28}, day{31}, day{30}, day{31}, day{30}, day{31}, day{31}, day{30}, day{31}, day{30}, day{31}};

```

Comment:
This is triggering the compiler bug that's breaking the Standard Library Header Units test. I'm reducing and reporting that, but it should be possible to perma-workaround this bug. (By "perma-workaround", I mean a change that's desirable on its own merits, so it doesn't need to be marked with a `TRANSITION` comment.)

Unrelated to the modules compiler bug, the issue is that if this function is ever called at runtime, MSVC will generate code to push each element of this array onto the stack. This is because distinct objects in C++ need distinct addresses. The compiler can avoid generating such code when it can prove to itself that the variable's address isn't observed by any code - however, arrays are virtually always address-taken and this defeats the optimizer (at this time). Therefore, in ordinary code, we use `static constexpr` for function-local arrays. (Function-local scalars can just be `constexpr`.) However, this is a `constexpr` function, so the compiler will complain that `static` variables aren't permitted. So, the proper workaround is:

* Move this outside of the function
* Make it `inline constexpr`
* Give it a more distinctive name than `_Days` (perhaps `_Last_day_table`)

That will make runtime codegen more efficient, and should avoid the modules bug (I haven't confirmed the latter with the full repro yet).

Reply:
Filed DevCom-1241697 "Standard Library Header Units: Local array with `day{31}` initializers emits 'not yet implemented' internal compiler error".